wiki/achievements.py

```diff
@@ -10,20 +13,25 @@ def process_file(f):
     with open(f) as file:
         for line in file:
             l = line.strip().split('\t')
-            if len(l) != 2:
+            if len(l) != 3:
                 continue
-            (k, v) = l
-            names[k] = v
+            (k, v, y) = l
+            names[k] = (v, y)
     return names
 
 
-REGEX_IN_DATE = r".*in\s*(?:[^ ,]*?)?\s*\d\d\d\d.*"
+REGEX_IN_DATE = r".*in\s*(?:[^ ,]*?)?\s*(\d\d\d\d).*"
 
 
-def process_page(id):
+def process_page(id, birth_year):
     page = wikipedia.page(pageid=id)
     in_date_regex = re.compile(REGEX_IN_DATE, re.IGNORECASE)
     tokenizer = nltk.data.load('tokenizers/punkt/english.pickle')
+
+    text_to_date = lambda x, y: datetime.strptime(x, y).date()
+
+    birth_date = text_to_date(birth_year, '%Y-%m-%d+%H:%M')
```

Comment:
However unlikely it might seem, that is correct. Feel free to confirm it yourself but I've tried it out and this is precisely the format it follows.